pandas/core/construction.py

```diff
@@ -562,7 +562,16 @@ def sanitize_array(
     if not is_list_like(data):
         if index is None:
             raise ValueError("index must be specified when data is not list-like")
-        data = construct_1d_arraylike_from_scalar(data, len(index), dtype)
+        if isinstance(data, str) and using_pyarrow_string_dtype():
+            from pandas.core.arrays.string_ import StringDtype
+
+            dtype = StringDtype("pyarrow_numpy")
+            data = dtype.construct_array_type()._from_sequence_of_strings(
```

Comment:
Can we not re-use `construct_1d_arraylike_from_scalar` and just pass in `StringDtype("pyarrow_numpy")`?

Reply:
done